src/libraries/System.Private.Uri/src/System/Uri.cs

```diff
@@ -1636,6 +1636,11 @@ public override bool Equals([NotNullWhen(true)] object? comparand)
                     return false;
             }
 
+            if (this.Scheme.Equals(UriSchemeMailto))
+            {
+                return this.ToString().Equals(obj.ToString());
```

Comment:
This condition is skipping a bunch of edge cases and fast paths.
Can you please move this down to before https://github.com/dotnet/runtime/blob/3fc0f5fd610f73f0397cf40f68bf6e13787a8f2c/src/libraries/System.Private.Uri/src/System/Uri.cs#L1732C9-L1734
and alter which `UriComponents` are selected for comparison.

Something along the lines of
```c#
UriComponents components = UriComponents.HttpRequestUrl;

if (_syntax.InFact(UriSyntaxFlags.MailToLikeUri))
{
    if (!obj._syntax.InFact(UriSyntaxFlags.MailToLikeUri))
        return false;

    components |= UriComponents.UserInfo;
}

string selfUrl = selfInfo.RemoteUrl ??= GetParts(components, UriFormat.SafeUnescaped);
string otherUrl = otherInfo.RemoteUrl ??= obj.GetParts(components, UriFormat.SafeUnescaped);
```

`GetHashCode` should also be updated to include the UserInfo when hashing.